internal/service/vmservice/bootstrap_test.go

```diff
@@ -422,9 +424,38 @@ func TestReconcileBootstrapDataMissingNetworkConfig(t *testing.T) {
 	require.True(t, conditions.GetReason(machineScope.ProxmoxMachine, infrav1alpha1.VMProvisionedCondition) == infrav1alpha1.WaitingForStaticIPAllocationReason)
 }
 
+func TestReconcileBootstrapDataFormatIgnition(t *testing.T) {
+	machineScope, mockClient, kubeClient := setupReconcilerTest(t)
+	vm := newVMWithNets("virtio=A6:23:64:4D:84:CB,bridge=vmbr0")
+	vm.VirtualMachineConfig.SMBios1 = biosUUID
+	machineScope.SetVirtualMachine(vm)
+
+	machineScope.ProxmoxMachine.Status.IPAddresses = map[string]infrav1alpha1.IPAddress{infrav1alpha1.DefaultNetworkDevice: {IPV4: "10.10.10.10"}}
+	createIP4AddressResource(t, kubeClient, machineScope, infrav1alpha1.DefaultNetworkDevice, "10.10.10.10")
+	createBootstrapSecretIgnition(t, kubeClient, machineScope)
+
+	data := "{\"ignition\":{\"config\":{},\"security\":{\"tls\":{}},\"timeouts\":{},\"version\":\"2.3.0\"},\"networkd\":{\"units\":[{\"contents\":\"[Match]\\nMACAddress=A6:23:64:4D:84:CB\\n\\n[Link]\\nName=eth0\\n\\n[Network]\\nAddress=10.10.10.10/24\\nGateway=10.10.10.11\\nDNS=1.2.3.4\\n\",\"name\":\"00-eth0.network\"}]},\"passwd\":{\"users\":[{\"name\":\"core\",\"sshAuthorizedKeys\":[\"ssh-ed25519 ...\"]}]},\"storage\":{\"files\":[{\"filesystem\":\"root\",\"path\":\"/etc/sudoers.d/core\",\"contents\":{\"source\":\"data:,core%20ALL%3D(ALL)%20NOPASSWD%3AALL%0A\",\"verification\":{}},\"mode\":384},{\"filesystem\":\"root\",\"overwrite\":true,\"path\":\"/etc/hostname\",\"contents\":{\"source\":\"data:,test\",\"verification\":{}},\"mode\":420},{\"filesystem\":\"root\",\"overwrite\":true,\"path\":\"/etc/proxmox-env\",\"contents\":{\"source\":\"data:,COREOS_CUSTOM_HOSTNAME=test%0ACOREOS_CUSTOM_INSTANCE_ID=41ec1197-580f-460b-b41b-1dfefabe6e32%0ACOREOS_CUSTOM_PROVIDER_ID=proxmox:%2F%2F41ec1197-580f-460b-b41b-1dfefabe6e32%0ACOREOS_CUSTOM_PRIVATE_IPV4=10.10.10.10%2F24\",\"verification\":{}},\"mode\":420}]},\"systemd\":{\"units\":[{\"contents\":\"[Unit]\\nDescription=kubeadm\\n# Run only once. After successful run, this file is moved to /tmp/.\\nConditionPathExists=/etc/kubeadm.yml\\nAfter=network.target\\n[Service]\\n# To not restart the unit when it exits, as it is expected.\\nType=oneshot\\nExecStart=/etc/kubeadm.sh\\n[Install]\\nWantedBy=multi-user.target\\n\",\"enabled\":true,\"name\":\"kubeadm.service\"},{\"enable\":true,\"name\":\"systemd-resolved.service\"}]}}"
```

Comment:
Feels hard to parse. To figure out what the issue is, if the test fails. I would prefer a multiline string like:

```
  data := `{
    "ignition": {
      "config": {},
      "security": {
        "tls": {}
      },
      "timeouts": {},
      "version": "2.3.0"
    },
    "networkd": {
      "units": [
        {
          "contents": "[Match]nMACAddress=A6:23:64:4D:84:CBnn[Link]nName=eth0nn[Network]nAddress=10.10.10.10/24nGateway=10.10.10.11nDNS=1.2.3.4n",
          "name": "00-eth0.network"
        }
      ]
    },
    "passwd": {
      "users": [
        {
          "name": "core",
          "sshAuthorizedKeys": [
            "ssh-ed25519 ..."
          ]
        }
      ]
    },
    "storage": {
      "files": [
        {
          "filesystem": "root",
          "path": "/etc/sudoers.d/core",
          "contents": {
            "source": "data:,core%20ALL%3D(ALL)%20NOPASSWD%3AALL%0A",
            "verification": {}
          },
          "mode": 384
        },
        {
          "filesystem": "root",
          "overwrite": true,
          "path": "/etc/hostname",
          "contents": {
            "source": "data:,test",
            "verification": {}
          },
          "mode": 420
        },
        {
          "filesystem": "root",
          "overwrite": true,
          "path": "/etc/proxmox-env",
          "contents": {
            "source": "data:,COREOS_CUSTOM_HOSTNAME=test%0ACOREOS_CUSTOM_INSTANCE_ID=41ec1197-580f-460b-b41b-1dfefabe6e32%0ACOREOS_CUSTOM_PROVIDER_ID=proxmox:%2F%2F41ec1197-580f-460b-b41b-1dfefabe6e32%0ACOREOS_CUSTOM_PRIVATE_IPV4=10.10.10.10%2F24",
            "verification": {}
          },
          "mode": 420
        }
      ]
    },
    "systemd": {
      "units": [
        {
          "contents": "[Unit]nDescription=kubeadmn# Run only once. After successful run, this file is moved to /tmp/.nConditionPathExists=/etc/kubeadm.ymlnAfter=network.targetn[Service]n# To not restart the unit when it exits, as it is expected.nType=oneshotnExecStart=/etc/kubeadm.shn[Install]nWantedBy=multi-user.targetn",
          "enabled": true,
          "name": "kubeadm.service"
        },
        {
          "enable": true,
          "name": "systemd-resolved.service"
        }
      ]
    }
  }`
```

Reply:
maybe have a collection of test data objects inside a const block

Reply:
this is just for mocking, it's not validating the data or something,
but just make sure the `reconcileBootstrapData()` works with a valid data.